### PR TITLE
Upgrade dropwizard.version to 2.1.0

### LINKED ISSUE #1462: upgrade dropwizard.version 2.1.0 or higher to fix CVEs
[Project board link](https://github.com/orgs/k8ssandra/projects/8/views/1?pane=issue&itemId=50559678)

Noticed bunch of CVEs during the vulnerability scanning:

They are all seem dependencies of the `dropwizard-core` dependency according to the `mvn dependency:tree`, to be able to fix these, we should bump `dropwizard-core` version to 2.1.0 or higher but it might require some code changes though since I was tried to apply patches to fix the CVEs but build was failed.

```
📦 amqp-client 5.16.0 (java-archive)
            Medium CVE-2023-46120 GHSA-mm8h-8587-p46h fixed in 5.18.0
        📦 jackson-databind 2.10.5.1 (java-archive)
            High CVE-2021-46877 GHSA-3x8x-79m2-3w2w fixed in 2.12.6
            High CVE-2020-36518 GHSA-57j2-w4cx-62h2 fixed in 2.12.6.1
            High CVE-2022-42003 GHSA-jjjh-jjxp-wpff fixed in 2.12.7.1
            High CVE-2022-42004 GHSA-rgv9-q543-rqg4 fixed in 2.12.7.1
        📦 jersey-common 2.33 (java-archive)
            Medium CVE-2021-28[16](https://github.com/wolfi-dev/os/actions/runs/7581390781/job/20649257298?pr=11430#step:7:17)8 GHSA-c43q-5hpj-4crv fixed in 2.34
        📦 jetty-http 9.4.50.v20221201 (java-archive)
            Medium CVE-2023-40167 GHSA-hmr7-m48g-48f6 fixed in 9.4.52
        📦 jetty-server 9.4.50.v20221201 (java-archive)
            Low CVE-2023-26049 GHSA-p26g-97m4-6q7c fixed in 9.4.51.v202302[17](https://github.com/wolfi-dev/os/actions/runs/7581390781/job/20649257298?pr=11430#step:7:18)
            Medium CVE-[20](https://github.com/wolfi-dev/os/actions/runs/7581390781/job/20649257298?pr=11430#step:7:21)23-26048 GHSA-qw69-rqj8-6qw8 fixed in 9.4.51.v20230[21](https://github.com/wolfi-dev/os/actions/runs/7581390781/job/20649257298?pr=11430#step:7:22)7
        📦 jetty-servlets 9.4.50.v20[22](https://github.com/wolfi-dev/os/actions/runs/7581390781/job/20649257298?pr=11430#step:7:23)1201 (java-archive)
            Low CVE-20[23](https://github.com/wolfi-dev/os/actions/runs/7581390781/job/20649257298?pr=11430#step:7:24)-36479 GHSA-3gh6-v5v9-6v9j fixed in 9.4.52

```

### PR DESCRIPTION
**Pull request by FieteO**
Bump `dropwizard.version` to `2.1.12`. Fixes #1462

### CODE REVIEW

**Comment by FieteO** [comment]
@adejanovski It looks like the setup of the tmate session is timing out. Is this related to my changes, or some flakiness in the Actions workflow?

**Comment by adejanovski** [comment]
> @adejanovski It looks like the setup of the tmate session is timing out. Is this related to my changes, or some flakiness in the Actions workflow?

tmate stays open for 30 minutes and then times out to allow the workflow to end.
Nothing misbehaving here, the core issue is in the breakage that the dropwizard upgrade created.

**Comment by FieteO** [comment]
Ah, I didn't see the failing test step before. Is there any way of running these (integration?) tests locally?

I tried to invoke `TEST_TYPE=ccm ./.github/scripts/run-tests.sh` but that fails with not finding the `ccm` executable.
I guess I could install that, but it would likely still fail right?

**Comment by FieteO** [comment]
`2.1.0` seems to work now. Shall I try to bump it to [the last patch version](https://mvnrepository.com/artifact/io.dropwizard/dropwizard-core) `2.1.12` @adejanovski ?

**Comment by adejanovski** [comment]
> `2.1.0` seems to work now. Shall I try to bump it to [the last patch version](https://mvnrepository.com/artifact/io.dropwizard/dropwizard-core) `2.1.12` @adejanovski ?

That seems like a good idea. Thanks for handling this @FieteO !

File: src/server/src/main/java/io/cassandrareaper/management/ClusterFacade.java [hunk 1]
```diff
@@ -181,7 +182,8 @@ public static CompactionStats parseCompactionStats(String json) throws IOExcepti
   private static <T> T parseJson(String json, TypeReference<T> ref) throws IOException {
     try {
       ObjectMapper mapper = new ObjectMapper();
-      mapper.registerModule(new Jdk8Module());
+      mapper.registerModule(new Jdk8Module()).registerModule(new JodaModule());
+      // ObjectMapper mapper = Jackson.newObjectMapper();
```
Comment (adejanovski) [inline src/server/src/main/java/io/cassandrareaper/management/ClusterFacade.java]: **nit:** could you remove this comment?

**Review by adejanovski — CHANGES_REQUESTED** [review]
Looking good :+1: 

Just one leftover comment to remove and we're good to go.

File: src/server/src/main/java/io/cassandrareaper/management/ClusterFacade.java (hunk 1 above)
Comment (FieteO) [reply src/server/src/main/java/io/cassandrareaper/management/ClusterFacade.java]: Done

**Review by adejanovski — APPROVED** [review]
Awesome! Thanks @FieteO 🙏

**Pull request merged**